### ISSUE TITLE
Make tedgeUrl optional when creating config_snapshot command

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2750]
**Is your feature improvement request related to a problem? Please describe.**

The `tedgeUrl` property of the `config_snapshot` command should not be mandatory when creating a local operation.

Having the `tedgeUrl` mandatory makes it hard for custom mappers to use a consistent tedgeUrl path, and it error prone as the component creating the command request needs to know about the file transfer service.

For example, the following MQTT message is required to create a valid `config_snapshot` command, note the `tedgeUrl` property.

```sh
tedge mqtt pub 'te/device/main///cmd/config_snapshot/local-1234' '{
    "status":"init",
    "type":"tedge-configuration-plugin",
    "tedgeUrl":"http://localhost:8005/tedge/file-transfer/config/tedge-configuration-plugin"
}' -r
```

**Describe the solution you'd like**

The following changes should be made
* the `tedgeUrl` should not be a mandatory property required by the tedge-agent when processing the `config_snapshot` command
* If the `tedgeUrl` property is not provided, the tedge-agent should set the `tedgeUrl` property itself.

After the above is implemented, it should be possible to create a `config_snapshot` using the following command:

```sh
tedge mqtt pub 'te/device/main///cmd/config_snapshot/local-1234' '{
    "status":"init",
    "type":"tedge-configuration-plugin"
}' -r
```

**Describe alternatives you've considered**

**Additional context**

**Comment by Ruadhri17** [comment]
Resolved by #2826. 

Ticket introduce optional `tedgeUrl` to `config_snapshot` payload. If a user sends message without `tedgeUrl`, it will be created by tedge-agent when switching from init to executing state. If somehow executing state will not contain `tedgeUrl` (for example someone will manually send executing message and won't put the url there), tedge agent do one more check before creating upload request and create the url if necessary. 

System test was created and can be found [here](https://github.com/thin-edge/thin-edge.io/blob/ac6323ad53a2e2ad2f555ee88666e5cb5209ccd0/tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot#L111).

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- [x] Test for ticket exists in the test suite.
- tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot
- [x] QA has tested the function and it's functioning according
